### PR TITLE
[Meshing] ParMmg - Pass MPI COMM world from modelpart data communicator

### PR DESCRIPTION
**Pull request by marcnunezc**
**📝 Description**
As requested in https://github.com/KratosMultiphysics/Kratos/pull/7808#discussion_r544188556, this passes the MPI_COM_WORLD to ParMmg from the data communicator. 

**🆕 Changelog**
- Pass MPI_COMM_WORLD from data comm.

### CODE REVIEW

File: applications/MeshingApplication/custom_utilities/parmmg/pmmg_utilities.cpp [hunk 1]
```diff
@@ -370,13 +372,13 @@ Element::Pointer ParMmgUtilities<PMMGLibrary::PMMG3D>::CreateFirstTypeElement(
...
 template<>
-void ParMmgUtilities<PMMGLibrary::PMMG3D>::InitMesh()
+void ParMmgUtilities<PMMGLibrary::PMMG3D>::InitMesh(const DataCommunicator& rDataCommunicator)
 {
     mParMmgMesh = nullptr;
 
     // We init the PMMG mesh and sol
     if (GetDiscretization() == DiscretizationOption::STANDARD) {
-        PMMG_Init_parMesh( PMMG_ARG_start, PMMG_ARG_ppParMesh, &mParMmgMesh, PMMG_ARG_pMesh, PMMG_ARG_pMet, PMMG_ARG_dim, 3, PMMG_ARG_MPIComm, MPI_COMM_WORLD, PMMG_ARG_end);
+        PMMG_Init_parMesh( PMMG_ARG_start, PMMG_ARG_ppParMesh, &mParMmgMesh, PMMG_ARG_pMesh, PMMG_ARG_pMet, PMMG_ARG_dim, 3, PMMG_ARG_MPIComm, MPIDataCommunicator::GetMPICommunicator(rDataCommunicator), PMMG_ARG_end);
```
Comment (philbucher) [inline applications/MeshingApplication/custom_utilities/parmmg/pmmg_utilities.cpp]: I recommend using the following checks too:
https://github.com/KratosMultiphysics/Kratos/blob/aa1ac92d4994e77950e180523a9ae7536a77b241/applications/CoSimulationApplication/mpi_extension/custom_python/co_simulation_mpi_extension.cpp#L34-L36

File: applications/MeshingApplication/custom_utilities/parmmg/pmmg_utilities.cpp [hunk 2]
```diff
@@ -378,7 +378,10 @@ void ParMmgUtilities<PMMGLibrary::PMMG3D>::InitMesh(const DataCommunicator& rDat
 
     // We init the PMMG mesh and sol
     if (GetDiscretization() == DiscretizationOption::STANDARD) {
-        PMMG_Init_parMesh( PMMG_ARG_start, PMMG_ARG_ppParMesh, &mParMmgMesh, PMMG_ARG_pMesh, PMMG_ARG_pMet, PMMG_ARG_dim, 3, PMMG_ARG_MPIComm, MPIDataCommunicator::GetMPICommunicator(rDataCommunicator), PMMG_ARG_end);
+        KRATOS_ERROR_IF_NOT(rDataCommunicator.IsDistributed()) << "ConnectMPI requires a distributed DataCommunicator!" << std::endl;
```
Comment (philbucher) [inline applications/MeshingApplication/custom_utilities/parmmg/pmmg_utilities.cpp]:
```suggestion
        KRATOS_ERROR_IF_NOT(rDataCommunicator.IsDistributed()) << "ParMMG requires a distributed DataCommunicator!" << std::endl;
```

Comment (marcnunezc) [reply applications/MeshingApplication/custom_utilities/parmmg/pmmg_utilities.cpp]: 🤦‍♂️ 

Thanks!

**Review by philbucher — APPROVED** [review]
:+1:

**Pull request merged**